mas_perception_libs/ros/src/mas_perception_libs/scene_detection_action.py

```diff
@@ -105,16 +106,17 @@ def _initialize(self, **kwargs):
         rospy.loginfo('will transform all poses to frame: ' + self._target_frame)
 
     def _execute_cb(self, _):
-        # subscribe and wait for cloud message TODO(minhnh) add timeout
-        self._cloud_sub = rospy.Subscriber(self._cloud_topic, PointCloud2, self._cloud_callback)
-        while self._cloud_msg is None:
-            continue
-
-        # stop subscribing to cloud topic to avoid overhead
-        self._cloud_sub.unregister()
-        # reset class field for next goal request
-        cloud_msg = self._cloud_msg
-        self._cloud_msg = None
+        # subscribe and wait for cloud message
+        try:
+            cloud_msg = rospy.wait_for_message(self._cloud_topic, PointCloud2, timeout=self._timeout)
+        except rospy.ROSInterruptException:
+            # shutdown event interrupts waiting for point cloud
+            raise
```

Comment:
do you have to raise a specific exception here?

Reply:
It'll just propagate `rospy.ROSInterruptException` up. I think this hits when you hit Ctrl + C on the terminal.